ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/app/AppContext.java

```diff
@@ -107,28 +99,11 @@ public void setFactory(Factory factory) {
         this.factory = factory;
     }
 
-    /**
-     * Adds passed project to list of opened projects
-     *
-     * @param descriptor
-     *         project descriptor which will be added
-     */
-    public void addOpenedProject(ProjectDescriptor descriptor) {
-        openedProjects.add(descriptor);
-    }
-
-    /**
-     * Removes passed project from list of opened projects
-     *
-     * @param descriptor
-     *         project descriptor which will be removed
-     */
-    public void removeOpenedProject(ProjectDescriptor descriptor) {
-        openedProjects.remove(descriptor);
+    public String getProjectsRoot() {
+        return projectsRoot;
     }
 
-    /** Returns list of opened projects. */
-    public List<ProjectDescriptor> getOpenedProjects() {
-        return openedProjects;
+    public void setProjectsRoot(String projectsRoot) {
```

Comment:
This project root that comes from machine dto at extension server startup, this isn't a projects root related to any project in project explorer